test/framework/kubernetesversions/data/debian_injection_script_control_plane.envsubst.sh

```diff
@@ -67,9 +70,12 @@ if [[ "$${KUBERNETES_VERSION}" != "" ]]; then
   echo "* testing CI version $${KUBERNETES_VERSION}"
   # Check for semver
   if [[ "$${KUBERNETES_VERSION}" =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
-    CI_URL="gs://kubernetes-release/release/$${KUBERNETES_VERSION}/bin/linux/amd64"
+    CI_URL="https://storage.googleapis.com/kubernetes-release/release/$${KUBERNETES_VERSION}/bin/linux/amd64"
     VERSION_WITHOUT_PREFIX="$${KUBERNETES_VERSION#v}"
-    DEBIAN_FRONTEND=noninteractive apt-get install -y apt-transport-https curl
+    export DEBIAN_FRONTEND=noninteractive
+    # sometimes the network is not immediately available, so we have to retry the apt-get update
+    retry 10 5 "apt-get update"
```

Comment:
I had to do this because the network did not immediately work after boot in our case. It's fine if we don't want this in the main repo, I can move it to a provider-specific script which is run before in CAPO.